### ISSUE TITLE
Element picker

### ISSUE THREAD

**Issue by streetwolf** [uBlockOrigin/uBlock-issues#2252]
### Prerequisites

- [X] I verified that this is not a filter issue (MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues))
- [X] This is not a [support issue or a question](https://old.reddit.com/r/uBlockOrigin/)
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue
- [X] The issue is not present after wholly disabling uBlock Origin ("uBO") in the browser
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### I tried to reproduce the issue when...

- [X] uBO is the only extension
- [X] uBO with default lists/settings
- [X] using a new, unmodified browser profile

### Description

Element picker problem at msn.com 

### A specific URL where the issue occurs

msn.com

### Steps to Reproduce

Go to msn.com
Select element picker
Try to block some element on the page

### Expected behavior

Element picker should work as expected.

### Actual behavior

The element picker highlights the entire page.  Selecting something to block results in the whole page being blocked.

EP works as expected on the latest version of MS Edge.

### uBlock Origin version

1.44.2

### Browser name and version

Firefox 105.0b9

### Operating System and version

Windows 11

**Comment by gothic-bum** [comment]
The Element Picker cannot find elements in a shadow root, see: https://github.com/uBlockOrigin/uBlock-issues/issues/2215

**Comment by gwarser** [comment]
Are you sure uBO is enabled on msn? Is cosmetic filtering enabled? Do you have any custom lists (can be caused by incorrectly hidden cookie dialog)?

**Comment by streetwolf** [comment]
I did lots of further testing.  EP worked fine on a test profile with only uBO as an addon.  This fact led to more testing on my normal profile and ultimately I deleted my cookie file and EP started working again.  However after a few restarts of Fx the EP problem came back.  At some further  point in my testing I've been able to have EP working consistently.  I noticed that along with the EP problem some elements weren't being blocked.  Now they are just like on my test profile.

So it appears my problem is cookie related.  I'm still running with a new cookie file and it's possible the problem might come back in time if the 'bad' cookies get saved.

**Comment by gorhill** [comment]
> ### I tried to reproduce the issue when...
> ...
>  - [x] using a new, unmodified browser profile

Then:

> I did lots of further testing. EP worked fine on a test profile with only uBO as an addon

Your original description of the issue led us to believe the issue was also present in a new, unmodified browser profile. Pointing out that the issue can't be reproduce in a new, unmodified browser profile helps better narrow the issue and saves time.

**Comment by streetwolf** [comment]
I think I found what's causing my problem with EP.  I use an addon called Forget Me Not (FMN) a cookie Manager. I've always set the options for cookie deletion for msn.com as Never which means take no action on cookies at this site.  When I delete the Never setting for the two URL's that are listed for msn.com PE works fine as well as blocking stuff that wasn't blocked before.  You would think telling the addon not to do anything with cookies at msn.com wouldn't cause a problem but it does.  Let it be known that I always had the site set to Never and didn't have any problems.  Something changed this is the past month or so I would say.

So to test this out install the addon FMN, go to msn.com and click on the FMN icon and you should see two URL's lists.  Click on the Create Rule button for each and select Never.  The site then should behave incorrectly, we hope.  Deleting the Never setting for both URL's should make things work as expected.

Of course your mileage may vary.

**Comment by krystian3w** [comment]
#2215

**Comment by krystian3w** [comment]
FMN = may no survive rewrite to manifest V3. Test with Auto Delete AutoCookie:  https://chrome.google.com/webstore/detail/cookie-autodelete/fhcgjolkccmbidfldomjliifgaodjagh

**Comment by streetwolf** [comment]
Installed Delete AutoCookie from the Firefox addon site.  It's the same version as the one on the Chrome store.  If I remember correctly things worked fine out of the box.  Once I started fooling around with the greylist and whitelist options I am back with the same problems.  I then deleted the expressions for msn.com thinking this would set the site back to an out of the box state but still have the problem.
